packages/delegation-process/test/revokeDelegation.test.ts

```diff
@@ -0,0 +1,266 @@
+import { randomUUID } from "crypto";
+import { fail } from "assert";
+import {
+  decodeProtobufPayload,
+  getMockDelegationProducer,
+  getRandomAuthData,
+  randomArrayItem,
+} from "pagopa-interop-commons-test";
+import {
+  Delegation,
+  DelegationId,
+  DelegationRevokedV2,
+  delegationState,
+  generateId,
+  TenantId,
+  fromDelegationV2,
+} from "pagopa-interop-models";
+import { describe, expect, it, vi } from "vitest";
+import { genericLogger } from "pagopa-interop-commons";
+import {
+  delegationNotFound,
+  delegationNotRevokable,
+  delegatorNotAllowToRevoke,
+} from "../src/model/domain/errors.js";
+import {
+  addOneDelegation,
+  delegationProducerService,
+  readDelegationEventByVersion,
+} from "./utils.js";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+const getNotRevocableStateSeeds = () => {
+  const rejectionOrRevokeDate = new Date();
+  rejectionOrRevokeDate.setMonth(new Date().getMonth() - 1);
+
+  return randomArrayItem([
+    {
+      delegationData: {
+        state: delegationState.rejected,
+        rejectedAt: rejectionOrRevokeDate,
+        rejectionReason: "Test is a test stop",
+      },
+      stamps: {
+        rejection: {
+          who: generateId<TenantId>(),
+          when: rejectionOrRevokeDate,
+        },
+      },
+    },
+    {
+      delegationData: {
+        state: delegationState.revoked,
+        revokedAt: rejectionOrRevokeDate,
+      },
+      stamps: {
+        revocation: {
+          who: generateId<TenantId>(),
+          when: rejectionOrRevokeDate,
+        },
+      },
+    },
+  ]);
+};
+
+describe("revoke delegation", () => {
+  it("should revoke a delegation if it exists", async () => {
+    const currentExecutionTime = new Date();
+    vi.useFakeTimers();
+    vi.setSystemTime(currentExecutionTime);
```

Comment:
These can be place in a `beforeAll`

Reply:
done in this [7605b94](https://github.com/pagopa/interop-be-monorepo/pull/1100/commits/7605b9460d09f6fae754a893a4818a0cb12ccfac)